psydac/api/ast/parser.py

```diff
@@ -535,20 +536,29 @@ def _visit_DefNode(self, expr, **kwargs):
             if isinstance(i, Shape):
                 inits.append(Assign(var, ZerosLike(i.arg)))
             else:
-                inits.append(Assign(var, Zeros(i)))
+                inits.append(Assign(var, Zeros(i, dtype=var.dtype)))
 
         inits.append(EmptyNode())
-        body =  tuple(inits) + body
-        name = expr.name
-        numpy_imports = ('array', 'zeros', 'zeros_like', 'floor')
-        math_imports  = (*self._math_functions,)
-        imports = [Import('numpy', numpy_imports)] + \
-                 ([Import('math', math_imports)] if math_imports else []) + \
-                  [*expr.imports]
-        results = [self._visit(a) for a in expr.results]
+        body          =  tuple(inits) + body
+        name          = expr.name
+
+        # If we are with complex object, we should import the mathematical function from numpy and not math to handle complex value.
+        if expr.domain_dtype=='complex':
+            numpy_imports = ('array', 'zeros', 'zeros_like', 'floor', *self._math_functions)
+            imports       = [Import('numpy', numpy_imports)] + \
+                            [*expr.imports]
+        # Else we import them from math
+        else:
+            math_imports  = (*self._math_functions,)
+            numpy_imports = ('array', 'zeros', 'zeros_like', 'floor')
+            imports       = [Import('numpy', numpy_imports)] + \
+                            ([Import('math', math_imports)] if math_imports else []) + \
+                            [*expr.imports]
```

Comment:
In the complex case you may import the functions from `cmath`. Could you please check if Pyccel supports them all?

Reply:
The `cmath` library is not supported by pyccel for now. Should I had a issue on the Pyccel page ?

Reply:
Yes please!

Reply:
The issue is created, for now, I have added a TODO to remember to change it in this commit [d707ec4](https://github.com/pyccel/psydac/pull/302/commits/d707ec4756b53630f270ad577b4c593ad199fd58)